Training/Training/Mealy.c

```diff
@@ -0,0 +1,108 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <stdio.h>
+#include <string.h>
+#include <windows.h>
+
+/// <summary> states of the Mealy machine
+/// A state transition diagram is attached for clear understanding </summary>
+typedef enum {
+   S0,  // Initial state
+   S1,  // After '0'
+   S2,  // After '01'
+   S3,  // After '011'
+   A1,  // After '1'
+   A2,  // After '11'
+   A3   // After '110'
+} State;
+
+void Mealy (char inpPath[], char outPath[]);
+State NextMealyState (State currentState, int input, int* output);
+
+void main (int argc, char* argv[]) {
+   char inpFilePath[MAX_PATH], outFilePath[MAX_PATH];
+   strcpy (inpFilePath, argv[1]);
+   strcpy (outFilePath, argv[2]);
+   Mealy (inpFilePath, outFilePath);
+}
+
+// Function to open input file and create a new file and store the result
+void Mealy (char inpPath[], char outPath[]) {
+   char ch, * tempPath = "temp_file.txt";
+   FILE* inpTestFile = fopen (inpPath, "r"), * outTestFile = fopen (outPath, "r"), * tempFile = fopen (tempPath, "w");
+   if (inpTestFile == NULL || outTestFile == NULL || tempFile == NULL) {
+      printf ("Error opening file");
+      return;
+   }
+   State currentState = S0;
+   int output = 0;
+   while ((ch = fgetc (inpTestFile)) != EOF) {

```

Comment:
Read full file at once, file should be read on calling function, not inside your state machine